File: Cpp_chunks/available_set.cpp
class SudokuGrid {
private:
    void defineAvailableSets(){
        SudokuCell *cell;
        for(int i: I){
            for(int j: I){
                cell = &grid[i][j];
                for(int candidate: Omega) {
                    if (!cell->hasValue() && !images->row[i]->has(candidate) && !images->col[j]->has(candidate) && !images->box[boxOf(i,j)]->has(candidate)){
                            cell->avSet->add(candidate);

                    }
                }
            }
        }
    }
public:
    SudokuGrid(vector<vector<int>> numberGrid, int n){
        /*...*/
        defineAvailableSets();
    }
};

File: Python/chunks/chapter_2/introduction/available_set.py
class SudokuGrid:
    """..."""
    def define_available_sets(self):
        for row in self.I:
            for col in self.I:
                cell = self.grid[row][col]
                if not cell.has_value():
                    for val in self.Omega:
                        if val in self.brackets.rowImage[row] or val in self.brackets.colImage[col] or val in self.brackets.boxImage[self.box_of(row, col)]:
                            cell.av_set.remove(val)

    def __init__(self, number_grid):
        """..."""
        self.define_available_sets()